scripts/sync-pkgs-wireit.js

```diff
@@ -0,0 +1,108 @@
+ * 2. Updates wireit config dependencies to match package.json dependencies.
+ *
+ * Note that this script does _not_ mutate:
+ * - victory-core
+ * - victory-vendor
+ * - victory-native
+ *
+ * If you are editing `victory-vendor` or `victory-native`, directly edit them.
+ * For **all other packages**, make your changes in `victory-core` first, test
+ * out, and then run this script to sync all the other packages.
+ */
+
+const fs = require("fs/promises");
+const path = require("path");
+const { log, error } = console;
+
+// ============================================================================
+// Config
+// ============================================================================
+const PKGS_ROOT = path.resolve(__dirname, "../packages");
+
+// Special packages
+const PKGS = {
+  CORE: "victory-core",
+  NATIVE: "victory-native",
+  VENDOR: "victory-vendor"
+};
+const PKGS_SET = new Set(Object.values(PKGS));
+
+// ============================================================================
+// Script
+// ============================================================================
+const cli = async ({ args = [] } = {}) => {
+  const workspaces = (await fs.readdir(PKGS_ROOT))
+    .filter((p) => p.startsWith("victory") && !PKGS_SET.has(p));
+
+  // Use the core package as the template for the rest.
+  const corePkg = require(`${PKGS_ROOT}/victory-core/package.json`);
+
+  for (let workspace of workspaces) {
+    const pkgPath = `${PKGS_ROOT}/${workspace}/package.json`;
+    const pkg = JSON.parse(await fs.readFile(pkgPath));
+
+    // Overwrite scripts and wireit configuration.
+    const scripts = JSON.parse(JSON.stringify(corePkg.scripts));
+    pkg.scripts = scripts;
+
+    // Start with wireit task config and add in dependencies
+    const wireit = JSON.parse(JSON.stringify(corePkg.wireit));
+    pkg.wireit = wireit;
+
+    const addDeps = (key, dep) => {
+      pkg.wireit[key].dependencies = pkg.wireit[key].dependencies || [];
+      pkg.wireit[key].dependencies.push(dep);
+    };
+
+    const crossDeps = Object.keys(pkg.dependencies).filter((p) => p.startsWith("victory"));
+    crossDeps.forEach((dep) => {
+      // Special case victory-vendor
+      if (dep === PKGS.VENDOR) {
+        [
+          "build:lib:esm",
+          "build:lib:cjs",
+          "build:dist:dev",
+          "build:dist:min",
+        ].forEach((key) => addDeps(key, `../${PKGS.VENDOR}:build`));
+        return;
+      }
+
+      // Normal case
+      // Make sure dependent libraries are built.
+      addDeps("build:lib:esm", `../${dep}:build:lib:esm`);
+      addDeps("build:lib:cjs", `../${dep}:build:lib:cjs`);
+
+      // Webpack depends on ESM output from other packages.
+      addDeps("build:dist:dev", `../${dep}:build:lib:esm`);
+      addDeps("build:dist:min", `../${dep}:build:lib:esm`);
+    });
+
+    // Hack: move sideEffects back to end
+    delete pkg.sideEffects;
+    pkg.sideEffects = false;
```

Comment:
I was thinking, if our `scripts` and `wireit` config are now generated, they should all be at the end.  So could we keep `sideEffects` "before the fold" if possible?

Reply:
Yeah! I should remove them and move them all way higher up (which is more typical)